Calculator - Human multiplication expressions

jjavierdguezas opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] **Closes:** #20187
- [x] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [x] **Tests:** Added/updated and all pass
- [x] **Localization:** All end user facing strings can be localized
- [x] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [x] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: [#4328](https://github.com/MicrosoftDocs/windows-dev-docs/pull/4328)

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

I took into account the combination of numbers (`num`), constants (`const`), functions (`func`) and expressions in parentheses (`(exp)`). The blank spaces between them were also considered. The solution is based on `regex`es. Some combinations were not handled as they are not common (but some could be handled if necessary)
- [x] `const const`: `pie = pi * e`  
- [ ] `const num`: `pi3 = pi * 3` not handled because it is not common
- [x] `const func`: `eln(100) = e * ln(100)`
- [x] `const (exp)`: `pi(1+1) = pi * (1+1)`
- [x] `num const`: `2pi = 2 * pi` 
- [ ] `num num`: not handled
- [x] `num func`: `2log10(100) = 2 * log10(100)`
- [x] `num (exp)`: `2(3+4) = 2 * (3+4)`
- [ ] `func const`: `log10(100)pi = log10(100) * pi` not handled because it is not common
- [ ] `func num`: `log10(100)3 = log10(100) * 3` not handled because it is not common
- [x] `func func`: `sin(pi)cos(pi) = sin(pi) * cos(pi)`
- [x] `func (exp)`: `log10(100)(2+3) = log10(100) * (2+3)`
- [ ] `(exp) const`: `(1+2)pi = (1+2) * pi` not handled because it is not common
- [ ] `(exp) num`: `(1+2)4 = (1+2) * 4` not handled because it is not common
- [x] `(exp) func`: `(1+1)cos(pi) = (1+1) * cos(pi)`
- [x] `(exp) (exp)`: `(1+1)(2+2) = (1+1) * (2+2)`

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

Commit:
fixes #20187

htcfreek commented on src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Calculator.UnitTest/QueryTests.cs:
```diff
@@ -113,5 +111,101 @@ public void NoErrorForDivisionByNumberWithDecimalDigits(string typedString)
+        [DataRow("pipipie", "pi * pi * pi * e")]
+        [DataRow("(1+1)(3+2)(1+1)(1+1)", "(1+1) * (3+2) * (1+1) * (1+1)")]
+        [DataRow("(1+1) (3+2)  (1+1)(1+1)", "(1+1) * (3+2) * (1+1) * (1+1)")]
+        public void RightHumaMultiplicationExpressionTransformation(string typedString, string expectedQuery)
```
Huma -> Human. (Same for the other checks.

jjavierdguezas replied:
solved

htcfreek reviewed (changes requested):
@jjavierdguezas 
I took a first fast look at the code. Found a small typo in all test method names.

Please fill the PR checklist. And it would be great if you can create a PR for our docs page later.

Commit:
handles PR reviews
- fix some typos
- updated dev docs
- added PR examples to tests
- improve method naming style

jjavierdguezas commented:
> @jjavierdguezas I took a first fast look at the code. Found a small typo in all test method names.
> 
> Please fill the PR checklist. And it would be great if you can create a PR for our docs page later.

Hi @htcfreek , if I didn't add new files, do I have to check the "`New binaries`" checkbox?

htcfreek commented:
> > @jjavierdguezas I took a first fast look at the code. Found a small typo in all test method names.
> > 
> > Please fill the PR checklist. And it would be great if you can create a PR for our docs page later.
> 
> Hi @htcfreek , if I didn't add new files, do I have to check the "`New binaries`" checkbox?

No need to check this box.

stefansjfw reviewed (approved):
LGTM. Thank you for your contribution!

jjavierdguezas committed:
Fix typo

Co-authored-by: Stefan Markovic <57057282+stefansjfw@users.noreply.github.com>
